buffer.py

```diff
@@ -79,18 +79,18 @@ def write_ulong(self, data: int):
         self.data += struct.pack("<Q", data)
 
     def write_string(self, data: str):
-        if (len(data) > 0):
+        if len(data) > 0:
             self.write_ubyte(0x0b)
             strlen = b""
-            value = len(data)
+            value = len(data.encode('utf-8'))

```

Comment:
Get the byte length of the characters instead of the characters length.

Reply:
oh shoot, thanks for looking into it for me, you've tested this already?

Reply:
Yeah. I can show my test code if you need

Reply:
oh alright, it's fine then, one of these days i'll have to write up some unit tests LOL